src/Installers/Windows/HostOptions/Product.wxs

```diff
@@ -0,0 +1,72 @@
+	<Product Id="$(var.ProductCode)" Name="$(var.ProductName)" Language="1033" Version="$(var.Version)" 
+			 Manufacturer="Microsoft Corporation" UpgradeCode="$(var.UpgradeCode)">
+		<Package InstallerVersion="$(var.InstallerVersion)" Compressed="yes" InstallScope="perMachine" />
+
+		<MajorUpgrade DowngradeErrorMessage="A newer version of [ProductName] is already installed." />
+		<Media Id="1" />
+
+        <!-- Custom Actions to assign a default of 0 to all switches -->
+        <CustomAction Id="Set_OPT_NO_ANCM" Property="OPT_NO_ANCM" Value="0" />
+        <CustomAction Id="Set_OPT_NO_FTS" Property="OPT_NO_FTS" Value="0" />
+        <CustomAction Id="Set_OPT_NO_LTS" Property="OPT_NO_LTS" Value="0" />
+        <CustomAction Id="Set_OPT_NO_SHAREDFX" Property="OPT_NO_SHAREDFX" Value="0" />
+        <CustomAction Id="Set_OPT_NO_RUNTIME" Property="OPT_NO_RUNTIME" Value="0" />
+        <CustomAction Id="Set_OPT_NO_X86" Property="OPT_NO_X86" Value="0" />
+        <CustomAction Id="Set_OPT_NO_SHARED_CONFIG_CHECK" Property="OPT_NO_SHARED_CONFIG_CHECK" Value="0" />
+
+        <InstallExecuteSequence>
+          <!-- Only execute the default set action if the property is undefined or was defined as an empty string -->
+          <Custom Action="Set_OPT_NO_ANCM" After="AppSearch">NOT OPT_NO_ANCM OR OPT_NO_ANCM=""</Custom>
+          <Custom Action="Set_OPT_NO_FTS" After="AppSearch">NOT OPT_NO_FTS OR OPT_NO_FTS=""</Custom>
+          <Custom Action="Set_OPT_NO_LTS" After="AppSearch">NOT OPT_NO_LTS OR OPT_NO_LTS=""</Custom>
+          <Custom Action="Set_OPT_NO_SHAREDFX" After="AppSearch">NOT OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=""</Custom>
+          <Custom Action="Set_OPT_NO_RUNTIME" After="AppSearch">NOT OPT_NO_RUNTIME OR OPT_NO_RUNTIME=""</Custom>
+          <Custom Action="Set_OPT_NO_X86" After="AppSearch">NOT OPT_NO_X86 OR OPT_NO_X86=""</Custom>
+          <Custom Action="Set_OPT_NO_SHARED_CONFIG_CHECK" After="AppSearch">NOT OPT_NO_SHARED_CONFIG_CHECK OR OPT_NO_SHARED_CONFIG_CHECK=""</Custom>
+        </InstallExecuteSequence>
+
+		<Feature Id="ProductFeature" Title="HostOptions" Level="1">
+			<ComponentGroupRef Id="ProductComponents" />
+		</Feature>
+	</Product>
+
+	<Fragment>
+		<Directory Id="TARGETDIR" Name="SourceDir">
+			<Directory Id="ProgramFilesFolder">
+        		<Directory Id="INSTALLFOLDER" Name="dotnet">
+    				<Directory Id="MM" Name="$(var.MajorVersion).$(var.MinorVersion)" />
+    			</Directory>
+      		</Directory>
+		</Directory>
+	</Fragment>
+
+	<Fragment>
+		<ComponentGroup Id="ProductComponents" Directory="MM">
+			<Component Id="OPT" KeyPath="yes" Guid="{5FF00E0C-2BEE-42CA-8359-D46D6CB74DD1}">
+				<RegistryKey Root="HKLM" Key="SOFTWARE\Microsoft\dotnet\host\options\$(var.MajorVersion).$(var.MinorVersion)">
+					<RegistryValue Name="OPT_NO_ANCM" Type="integer" Value="[OPT_NO_ANCM]" />
+				</RegistryKey>
+				<RegistryKey Root="HKLM" Key="SOFTWARE\Microsoft\dotnet\host\options\$(var.MajorVersion).$(var.MinorVersion)">
```

Comment:
Put the registry key someplace central to define it once and use it everywhere else